python/cudf/cudf/testing/_utils.py

```diff
@@ -333,6 +333,33 @@ def xfail_param(param, **kwargs):
     return pytest.param(param, marks=pytest.mark.xfail(**kwargs))
 
 
+def assert_column_memory_eq(
+    lhs: cudf.core.column.ColumnBase, rhs: cudf.core.column.ColumnBase
+):
+    """Assert the memory location and size of `lhs` and `rhs` are equivalent.
+
+    Both data pointer and mask pointer are checked. Also recursively check for
+    children to the same contarints. Also fails check if the number of children
+    mismatches at any level.
+    """
+    assert lhs.base_data_ptr == rhs.base_data_ptr
+    assert lhs.base_mask_ptr == rhs.base_mask_ptr
+    assert lhs.size == rhs.size
```

Comment:
Do we want to check that the `base_size` and `offset` are independently different instead? Otherwise, I think `s[1:5]._column` and `s[2:6]._column` will compare equal.

Reply:
Good catch.

Reply:
I think checking `size` is also important. Otherwise `s[0:1]` and `s` will compare equal too.

Reply:
I've added a few test cases for this test utilities so that we are not missing corner cases like this.